firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -127,17 +127,27 @@ class FieldPath(object):
         parts: (one or more strings)
             Indicating path of the key to be used.
     """
-    pattern = re.compile(r'[A-Za-z_][A-Za-z_0-9]*')
+    simple_field_name = re.compile(r'[A-Za-z_][A-Za-z_0-9]*')
 
     def __init__(self, *parts):
         for part in parts:
-            if not isinstance(part, str):
-                raise ValueError("One or more components is not a string")
+            try:
+                string = basestring
+            except NameError:
+                string = str
+            if not isinstance(part, string):
+                raise ValueError("One or more components is not a string.")
         self.parts = tuple(parts)
 
-    @classmethod
-    def from_string(cls, string):
-        """ Creates a FieldPath with a string representation.
+    @staticmethod
+    def from_string(string):
+        """ Creates a FieldPath from an unicode string representation.
```

Comment:
" a unicode", not "an unicode". Stupid English language.

Reply:
(I think it's because "unicode" is pronounced as if it had an initial consonant: "yoonicode".)